probe/proc/reader_linux.go

```diff
@@ -0,0 +1,203 @@
+package proc
+
+import (
+	"bytes"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/bluele/gcache"
+)
+
+const (
+	filesCacheLen        = 512
+	filesCacheExpiration = 60 * time.Second
+)
+
+var tcpFiles = []string{
+	"net/tcp",
+	"net/tcp6",
+}
+
+// A cache for files handles
+// Note: not intended to be used from multiple goroutines
```

Comment:
Ok, but we are adding a mutex for something that will probably never be used concurrently... but not a big deal.